src/Controller/EnMarche/EventManagerController.php

```diff
@@ -189,4 +165,50 @@ public function contactMembersAction(Request $request, Event $event): Response
             'form' => $form->createView(),
         ]);
     }
+
+    /**
+     * @Route("/inscrits/imprimer", name="app_event_print_members")
+     * @Method("POST")
+     */
+    public function printMembersAction(Request $request, Event $event): Response
+    {
+        $registrations = $this->getRegistrations($request, $event, self::ACTION_PRINT);
+
+        if (!$registrations) {
+            return $this->redirectToRoute('app_event_members', [
+                'slug' => $event->getSlug(),
+            ]);
+        }
+
+        $html = $this->renderView('events/print_members.html.twig', [
+            'registrations' => $registrations,
+        ]);
+
+        return new PdfResponse($this->getPdfForResponse($html), 'Liste des participants.pdf');
+    }
+
+    private function getRegistrations(Request $request, Event $event, string $action): array
```

Comment:
Please move this method in the `EventRegistrationManager` instead with a `handleActionRequest` and constants there, should be public to be usable in the view. If the param is not a valid constant a 400 should be return thanks to a proper `InvalidEventRegistrationActionException`.

Reply:
@HeahDude ok for the oublic constants, but I'm not sure that method moving is useful. This method executes validation of token and request parameters and if they are good it retrieves the registrations by simple calling of `findByUuidAndEvent`.